engine/src/main/java/org/destinationsol/common/SolRandom.java

```diff
@@ -0,0 +1,171 @@
+     * @param max  a positive value
+     */
+    public static int seededRandomInt(float perc, int max) {
+        int min = (int) (max * perc);
+        if (min == max) {
+            throw new AssertionError("intRnd min equals max " + min);
+        }
+        return seededRandom.nextInt(max - min) + min;
+    }
+
+    /**
+     * Returns a seeded random int v such that min <= v && v <= max
+     */
+    public static int seededRandomInt(int min, int max) {
+        return seededRandom.nextInt(max - min) + min;
+    }
+    
+    /**
+     * Returns a random float v such that -minMax <= v && v < minMax
+     *
+     * @param minMax a positive value
+     */
+    public static float randomFloat(float minMax) {
+        return SolRandom.randomFloat(-minMax, minMax);
+    }
+
+    /**
+     * Returns a random float v such that min <= v && v < max. Min shouldn't equal to max
+     */
+    public static float randomFloat(float min, float max) {
+        float result = max;
+        if (min == max) {
+            Gdx.app.log("SolMath", "rnd was called with bad parameters! Min " + min + " matches max " + max + ", accepting max.");
+            Gdx.app.log("SolMath", "Please review appropriate code in the stack dump:");
+            for (StackTraceElement ste : Thread.currentThread().getStackTrace()) {
+                Gdx.app.log("SolMath", ste.toString());
+            }
+    
+            return result;
+        }
+        return random.nextFloat() * (max - min) + min;
+    }
+
+    /**
+     * Returns a random int v such that 0 <= v && v < max.
+     *
+     * @param max a positive value
+     */
+    public static int randomInt(int max) {
+        return SolRandom.randomInt(0f, max);
+    }
+
+    /**
+     * Returns a random int v such that max*perc <= v && v < max.
+     *
+     * @param perc should be >= 0 and < 1
+     * @param max  a positive value
+     */
+    public static int randomInt(float perc, int max) {
+        int min = (int) (max * perc);
+        if (min == max) {
+            throw new AssertionError("intRnd min equals max " + min);
+        }
+        return random.nextInt(max - min) + min;
+    }
+
+    /**
+     * Returns a random int v such that min <= v && v <= max
+     */
+    public static int randomInt(int min, int max) {
+        return random.nextInt(max - min) + min;
+    }
+
+    /**
+     * @return a random element of a list
+     */
+    public static <T> T randomElement(List<T> list) {
+        int idx = randomInt(list.size());
+        return list.get(idx);
+    }
+    
+    /**
+     * @return a seeded random element of a list
+     */
+    public static <T> T seededRandomElement(List<T> list) {
+        int idx = seededRandomInt(list.size());
+        return list.get(idx);
+    }
+
+    public static void setSeed(long seed) {
+        SolRandom.seed = seed; 
+        seededRandom = new Random(seed);
+    }
+
+    /**
+     * Gets the seed currently in use
+     * @return The seed from seededRandom
```

Comment:
IMO, the javadoc should not mentioned 'seededRandom` since that's an implementation detail, and not an interface one.